libbeat/outputs/elasticsearch/config.go

```diff
@@ -24,9 +24,12 @@ type elasticsearchConfig struct {
 }
 
 type Template struct {
-	Name      string `config:"name"`
-	Path      string `config:"path"`
-	Overwrite bool   `config:"overwrite"`
+	Enabled           bool   `config:"enabled"`
+	Name              string `config:"name"`
+	IgnoreES2xVersion bool   `config:"ignore_es2x_version"`
+	Path              string `config:"path"`
+	PathES2x          string `config:"path_es2x"`
+	Overwrite         bool   `config:"overwrite"`
```

Comment:
I went for something in between:

```
  # If set to true, filebeat checks the Elasticsearch version at connect time, and if it
  # is 2.x, it loads the file specified by the template.versions.2x.path setting. The 
  # default is true.
  template.versions.2x.enabled: true

  # Path to the Elasticsearch 2.x version of the template file.
  template.versions.2x.path: "filebeat.template-es2x.json"
```

Not as flexible as your version, but pretty easily extensible in the future. Also, I think the flexibility is not needed, as you generally don't want a Beat to talk to multiple ES versions. This is just for making the getting starting experience easier.